utxo_verification.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+)
+
+
+
+func ReverseBytes(data []byte) []byte {
+	for i, j := 0, len(data)-1; i < j; i, j = i+1, j-1 {
+		data[i], data[j] = data[j], data[i]
+	}
+	return data
+}
```

Comment:
There is already a function for this: https://pkg.go.dev/slices@go1.23.2#Reverse -- see example